packages/ckeditor5-dev-env/lib/translations/collect-utils.js

```diff
@@ -0,0 +1,202 @@
+
+const ckeditor5PackagesDir = path.join( process.cwd(), 'packages' );
+const langContextSuffix = path.join( 'lang', 'contexts.json' );
+const corePackageName = 'ckeditor5-core';
+
+const utils = {
+	/**
+	 * Collect translations and returns array of translations.
+	 *
+	 * @returns {Array<Object>}
+	 */
+	collectTranslations() {
+		const srcPaths = [ process.cwd(), 'packages', '*', 'src', '**', '*.js' ].join( '/' );
+
+		const files = glob.sync( srcPaths )
+			.filter( ( srcPath ) => !srcPath.match( /packages\/[^\/]+\/src\/lib\// ) );
+
+		const translations = [];
+
+		for ( const filePath of files ) {
+			const content =  fs.readFileSync( filePath, 'utf-8' );
+
+			translations.push( ...utils._getTranslationCallsFromFile( filePath, content ) );
+		}
+
+		return translations;
+	},
+
+	_getTranslationCallsFromFile( filePath, fileContent ) {
+		const fullMatches = fileContent.match( / t\([^)]+?\)/gm ) || [];
+
+		return fullMatches.map( ( fullMatch ) => {
+			const stringMatch = fullMatch.match( /'([^']+?)'/ );
+
+			if ( !stringMatch ) {
+				logger.error( `Invalid translation call: ${ fullMatch } in ${ filePath }` );
+
+				return;
+			}
+
+			const key = stringMatch[ 1 ];
+
+			const contextMatch = key.match( /\[context: ([^\]]+)\]/ );
+			const sentenceMatch = key.match( /^[^\[]+/ );
+			const packageMatch = filePath.match( /\/(ckeditor5-[^\/]+)\// );
+
+			return {
+				filePath,
+				key,
+				package: packageMatch[ 1 ],
+				context: contextMatch ? contextMatch[ 1 ] : null,
+				sentence: sentenceMatch[ 0 ].trim(),
+			};
+		} )
+		.filter( translationCall => !!translationCall );
+	},
+
+	/**
+	 * Traverse all packages and returns Map of the all founded language contexts informations (file content and file name).
+	 *
+	 * @returns {Map<String, Object>}
+	 */
+	getContexts() {
+		const mapEntries = utils.getPackagesContainingContexts().map( packageName => {
+			const pathToContext = path.join( ckeditor5PackagesDir, packageName, langContextSuffix );
+
+			return [ packageName, {
+				filePath: pathToContext,
+				content: JSON.parse( fs.readFileSync( pathToContext, 'utf-8' ) )
+			} ];
+		} );
+
+		return new Map( mapEntries );
+	},
+
+	getPackagesContainingContexts() {
+		return fs.readdirSync( ckeditor5PackagesDir )
+			.filter( ( packageName ) => fs.existsSync(
+				path.join( ckeditor5PackagesDir, packageName, langContextSuffix )
+			) );
+	},
+
+	/**
+	 * @param {Map.<Object>} contexts Map of the language contexts.
+	 * @param {Array.<Object>} translations Array of the translations.
+	 * @returns {Array<String>}
+	 */
+	getMissingContextErrorMessages( contexts, translations ) {
+		const errors = [];
+
+		for ( const translation of translations ) {
+			const errorMessage = utils.maybeGetContextErrorMessage( contexts, translation );
+
+			if ( errorMessage ) {
+				errors.push( errorMessage );
+			}
+		}
+
+		return errors;
+	},
+
+	maybeGetContextErrorMessage( contexts, translation ) {
```

Comment:
Odd function. Couldn't be something like `validateContexts()` which logs the error or does something else (can return it too if it must)?